### PR TITLE
[webgpu] Add resizeNearestNeighbor support

### PR DESCRIPTION
**Pull request by haoyunfeix**
FEATURE

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5059)

### CODE REVIEW

**Comment by haoyunfeix** [comment]
@qjia7 @axinging @xhcao @gyagp PTAL

File: tfjs-backend-webgpu/src/kernels/resize_nearest_neighbor_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  alignCorners: boolean;
+  halfPixelCenters: boolean;
+  oldHeight: number;
+  oldWidth: number;
+
+  constructor(
+      inputShape: [number, number, number, number], newHeight: number,
+      newWidth: number, alignCorners: boolean, halfPixelCenters: boolean) {
+    this.oldHeight = inputShape[1];
+    this.oldWidth = inputShape[2];
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/resize_nearest_neighbor_webgpu.ts:38]: Remove `this.oldHeight` and `this.oldWidth`? And line 31, 32

Comment (haoyunfeix) [reply tfjs-backend-webgpu/src/kernels/resize_nearest_neighbor_webgpu.ts]: Thanks for pointing this, fixed.

**Review by qjia7 — APPROVED** [review]
LGTM. @lina128 @jinjingforever please take a look, thanks.

**Pull request merged**